experimental/rockbuilder/rockbuilder.py

```diff
@@ -142,6 +154,16 @@ def get_build_arguments(rock_builder_home_dir):
         args.hipify = True
 
     # add output dir to environment variables
+    if args.project and args.src_dir:
+        # single project case with optional src_dir specified
+        parent_dir = args.src_dir.parent
+        if parent_dir == args.src_dir:
+            print("Error, --src-dir parameter is not allowed to be a root-directory")
+            sys.exit(1)
```

Comment:
I just found out about `parser.error`, which you can use here: https://docs.python.org/3/library/argparse.html#exiting-methods

```suggestion
        if parent_dir == args.src_dir:
            parser.error("--src-dir parameter is not allowed to be a root-directory"
```

That will print the usage info and exit for you.